docs/sphinx/manual/building_hpx.rst

```diff
@@ -563,353 +198,123 @@ Available build types are:
-you used in your installation.
-
-Build |hpx|, finally
-....................
-
-.. code-block:: bash
-
-   cd /path/to
-   git clone https://github.com/STEllAR-GROUP/hpx.git
-   mkdir build-hpx && cd build-hpx
-
-To build with Clang, execute:
-
-.. code-block:: bash
-
-   cmake ../hpx \
-       -DCMAKE_CXX_COMPILER=clang++ \
-       -DBOOST_ROOT=/path/to/boost \
-       -DHWLOC_ROOT=/path/to/hwloc \
-       -DHPX_WITH_GENERIC_CONTEXT_COROUTINES=On
-   make -j
-
-For more detailed information about using |cmake|, please refer its documentation
-and to the section :ref:`building_hpx`.
-
-Alternative installation method of |hpx| on OS X (Mac)
-......................................................
-
-Alternatively, you can install a recent version of gcc as well as all
-required libraries via MacPorts:
-
-#. Install |macports|
-
-#. Install CMake, gcc, hwloc:
-
-   .. code-block:: bash
-
-      sudo brew install cmake
-      sudo brew install boost
-      sudo brew install hwloc
-      sudo brew install make
-
-#. You may also want:
-
-   .. code-block:: bash
-
-      sudo brew install gperftools
+    .. note::
+       Please pay special attention to the section about :option:`HPX_WITH_MALLOC:STRING` as this is crucial for getting decent performance.
 
-#. If you need to build Boost manually (the Boost package of MacPorts is built
-   with Clang, and unfortunately doesn't work with a GCC-build version of |hpx|):
+   .. important::
 
-   .. code-block:: bash
+       If you are building |hpx| for a system with more than 64 processing units,
+       you must change the |cmake| variable ``HPX_WITH_MAX_CPU_COUNT`` (to a value at least as big as the
+       number of (virtual) cores on your system). Note that the default value is 64.
 
-      wget https://dl.bintray.com/boostorg/release/1.69.0/source/boost_1_69_0.tar.bz2
-      tar xjf boost_1_69_0.tar.bz2
-      pushd boost_1_69_0
-      export BOOST_ROOT=$HOME/boost_1_69_0
-      ./bootstrap.sh --prefix=$BOOST_DIR
-      ./b2 -j8
-      ./b2 -j8 install
-      export DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$BOOST_ROOT/lib
-      popd
+   .. caution::
 
-#. Build |hpx|:
+       Compiling and linking |hpx| needs a considerable amount of memory. It is
+       advisable that at least 2 GB of memory per parallel process is available.
 
-   .. code-block:: bash
+   .. note::
 
-      git clone https://github.com/STEllAR-GROUP/hpx.git
-      mkdir hpx-build
-      pushd hpx-build
-      export HPX_ROOT=$HOME/hpx
-      cmake -DCMAKE_CXX_COMPILER=g++ \
-          -DCMAKE_CXX_FLAGS="-Wno-unused-local-typedefs" \
-          -DBOOST_ROOT=$BOOST_ROOT \
-          -DHWLOC_ROOT=/opt/local \
-          -DCMAKE_INSTALL_PREFIX=$HOME/hpx \
-          -DHPX_WITH_GENERIC_CONTEXT_COROUTINES=On \
-               $(pwd)/../hpx
-      make -j8
-      make -j8 install
-      export DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$HPX_ROOT/lib/hpx
-      popd
+       Many Linux distributions use ``make`` as an alias for ``gmake``.
```

Comment:
I think we agreed to remove this in the chat no? or maybe I don't recall correctly ^^